### PR TITLE
 #2750 fix for font saturation on domain review and summary pages- [AD]

### LINKED ISSUE #2750: Contact Font Saturation Needs To Be Fixed Within
### Current Behavior

Currently when we have more than one Contact in the "Other employees from your organization" in both the review page when you submit a domain request, and when you have a domain request in review state it displays as this grey font that is almost transparent with a white background.

See screenshot below for what it currently looks like 

### Expected Behavior

Contacts are in [this Figma spec here](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-registar?node-id=2703-31376&node-type=canvas&t=Sd5vKuWCqh2BIwoO-0) so it's viewable.

Screenshot 2024-09-19 at 3 53 59 PM

### Steps to Reproduce

1. Create a domain request and in the "Other employees from your organization" section, add more than one contact. When you get the last "review" page part, you'll see the font is wonky
2. Same for if a domain request is in the "in review" state and you have more than 1 contact 

### Environment

_No response_

### Additional Context

Screenshot 2024-09-13 at 8 42 39 AM
Screenshot 2024-09-13 at 8 44 31 AM

### Issue Links

_No response_

### PR DESCRIPTION
**Pull request by asaki222**
## Ticket

Resolves #2750 

## Changes

-  removed the definition list, and replaced it with an unorder lists
- added styling to contact to make it more accessible

## Context for reviewers

The changes on this should reflect the figma design linked in the issue. 

#### Satisfied acceptance criteria and met development standards

- [x] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Review by AnnaGingle — APPROVED** [review]
Great work!

File: src/registrar/templates/includes/summary_item.html [hunk 1]
```diff
@@ -32,16 +28,23 @@ <h4 class="register-form-review-header">{{ sub_header_text }}</h4>
             {% include "includes/contact.html" with contact=value|first %}
           {% else %}
             {% if value %}
-              <dl class="usa-list usa-list--unstyled margin-top-0">
+              <ul class="usa-list usa-list--unstyled margin-top-0">
```
Comment (therealslimhsiehdy) [inline src/registrar/templates/includes/summary_item.html:31]: **Question:** Could you elaborate on why we're using ul instead of dl in this case? :O

Comment (asaki222) [reply src/registrar/templates/includes/summary_item.html:31]: I was messing around with the styling, but I can revert!

Comment (asaki222) [reply src/registrar/templates/includes/summary_item.html]: Reverted!

**Review by therealslimhsiehdy — APPROVED** [review]
LGTM! Screenshot 2024-09-20 at 12 51 13 PM

**Pull request merged**